test/Main.hs

```diff
@@ -100,18 +95,23 @@ parseResponses :: TestTree
       , (JSON.object [("foo", JSON.object empty)], "Invalid status code pattern: \"foo\"")
       ]
 
-    mkAccepted :: (JSON.Value, Paths.Responses) -> TestTree
-    mkAccepted (input, expected)
-      = testCase (convertText ("accepts: " <> show input))
-      $ pure expected @=? parseJSON input
-
-    mkRejected :: (JSON.Value, Text) -> TestTree
-    mkRejected (input, expectedMessage)
-      = testCase (convertText ("rejects: " <> show input))
-      $ Left ("Error in $: " <> convertText expectedMessage) @=? parseJSON input
-
-    parseJSON :: JSON.Value -> Either String Paths.Responses
-    parseJSON = JSON.parseEither JSON.parseJSON
-
 main :: IO ()
 main = defaultMain suite
+
+mkAccepted
+  :: (Eq a, JSON.FromJSON a, Show a)
+  => (JSON.Value, a)
+  -> TestTree
+mkAccepted (input, expected)
```

Comment:
Why does this and `mkRejected` take a tuple instead of two arguments?

Reply:
Because `HashMap.toList` returns a list of tuples and I apply a functor over it. I could also use `uncurry` at the call side.

But `uncurry` was not in scope, so I decided to use syntax over the function.

Reply:
I think its best to keep it a tuple, as we have 2 call sides where we'd have to add `uncurry`. So unpacking here is less total AST.